java/src/main/java/com/adhoc/flight/client/AdhocFlightClient.java

```diff
@@ -192,70 +190,134 @@ public FlightInfo getInfo(String query, CallOption ... options) {
     public FlightStream getStream(FlightInfo flightInfo, CallOption... options) {
         return client.getStream(flightInfo.getEndpoints().get(0).getTicket(), options);
     }
-
+    
     /**
      * Make FlightRPC requests to the Dremio Flight Server Endpoint to retrieve results of the
      * provided SQL query.
      *
      * @param query the SQL query to execute.
      * @param headerCallOption client properties to execute provided SQL query with.
-     * @param rowProcessor object to use for processing rows
-     * @throws Exception if error occurs during query execution.
+     * @param fileToSaveTo the file to which the binary data of the resulting {@link VectorSchemaRoot}
+     *        should be saved.
+     * @param printToConsole whether or not the query results should be printed to the console.
+     * @throws Exception if an error occurs during query execution.
      */
-    public void runQuery(String query, HeaderCallOption headerCallOption, RowProcessor rowProcessor) throws Exception {
-        // Note: Dremio client property "schema" can be set for Flight RPC request to the server.
-        //       The "schema" property provides context for a query. For instance, the catalog or schema
-        //       of the query can be provided, so that the query does not have to reference the full dataset path.
+    public void runQuery(String query,
+                         HeaderCallOption headerCallOption,
+                         File fileToSaveTo, boolean printToConsole) throws Exception {
 
-        //       Below is a code snippet demonstrating how the "schema" client property can be set.
-        //final Map<String, String> schemaProperty = ImmutableMap.of(
-        //        "schema", "test-schema");
-        //final CallHeaders callHeaders = new FlightCallHeaders();
-        //properties.forEach(callHeaders::insert);
-        //final HeaderCallOption routingCallOptions = new HeaderCallOption(callHeaders);
+        FlightInfo flightInfo = getInfo(query, bearerToken, headerCallOption);
```

Comment:
final marker here and in the try with resources.

Reply:
Added the `final` modifier, however I can't put that on a try-with-resources block, since `FlightInfo` is not an `AutoCloseable`.